sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -49,11 +49,11 @@ static int max98373_hw_params(struct snd_pcm_substream *substream,
 	for_each_rtd_codec_dais(rtd, j, codec_dai) {
 		if (!strcmp(codec_dai->component->name, MAX_98373_DEV0_NAME)) {
 			/* DEV0 tdm slot configuration */
-			snd_soc_dai_set_tdm_slot(codec_dai, 0x03, 3, 8, 24);
+			snd_soc_dai_set_tdm_slot(codec_dai, 0x03, 3, 8, 32);
 		}
 		if (!strcmp(codec_dai->component->name, MAX_98373_DEV1_NAME)) {
 			/* DEV1 tdm slot configuration */
-			snd_soc_dai_set_tdm_slot(codec_dai, 0x0C, 3, 8, 24);
+			snd_soc_dai_set_tdm_slot(codec_dai, 0x0C, 3, 8, 32);
```

Comment:
the previous commit bc7477fc2ab4882ee20cc58a514f6991c1dfc636 ('ASoC: Intel: Boards: tgl_max98373: Update TDM configuration in hw_params') already increase the slot width from 16 to 24, now we are doing the same from 24 to 32 using the same argument that this follows the topology definition.

We'd need a clearer explanation here, did the topology file change twice?

Reply:
also need a Fixes: tag so that it's back-ported to stable kernels.

Reply:
> the previous commit [bc7477f](https://github.com/thesofproject/linux/commit/bc7477fc2ab4882ee20cc58a514f6991c1dfc636) ('ASoC: Intel: Boards: tgl_max98373: Update TDM configuration in hw_params') already increased the slot width from 16 to 24, now we are doing the same from 24 to 32 using the same argument that this follows the topology definition.
> 
> We'd need a clearer explanation here, did the topology file change twice?


This is the 1st time, ODM is cross checking THD. Reason is: in earlier version of DUT, it was not right speakers used.

During this issue debug, team observed: BCLK is reflecting 32-bit, but this configuration here is otherwise. So, there is mismatch.
In earlier topology versions, BCLK was set to: 9.6Mhz (25 x 48 x 8,  24/25 valid bits). As the customers ask was to change to 32-bit, it got changed in tplg. Yes, there was a change in topology.

Reply:
> also need a Fixes: tag so that it's back-ported to stable kernels.

Added Fixes TAG now. Please review.

Reply:
the commit message should really be something like:

The bclk configuration was changed in the topology file to be based on 12.288MHz and the hw_params is based on the s32le format, but the the TDM slot size remained set to 24. This inconsistency creates audible noise and needs to be corrected.

Reply:
Thank you @plbossart for providing help text to update in description. corrected now.